### PR TITLE
Change private method to protected

### PR DESCRIPTION
**Pull request by tharakawijekoon**
Change the private method to protected so that classes that extend the X509CertificateAuthenticator can use the same method.

### CODE REVIEW

**Comment by tharindu-b-hewage** [comment]
Hi @tharakawijekoon,

Could you please kindly explain more about the insights behind this design change, and what benefits would it bring to the authenticator implementation?

**Comment by tharakawijekoon** [comment]
Hi @tharindu-bandara,

I believe a derived class could benefit from using the method, making the method `protected` would allow it to do so, while making it `private` would deny it that benefit.

Yes, Keeping the method `private` would allow that future version to change the behavior (or eliminate the method entirely), while making it `protected` would require all future versions of the class to keep the same behavior, thus denying the benefit that could be reaped from changing it.

We have a realistic scenario here[1] where a derived class will benefit from being able to access the method.

[1]https://github.com/tharakawijekoon/CustomX509CertificateAuthenticator

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java [hunk 1]
```diff
@@ -424,7 +424,7 @@ protected boolean retryAuthenticationEnabled() {
      * @param cert                  x509 certificate.
      * @param authenticationContext authenticationContext
      */
-    private String getMatchedAlternativeName(X509Certificate cert, AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+    protected String getMatchedAlternativeName(X509Certificate cert, AuthenticationContext authenticationContext) throws AuthenticationFailedException {
```
Comment (piraveena) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java]: This line seems to exceed 120 character length

Comment (piraveena) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java]:
```suggestion
    protected String getMatchedAlternativeName(X509Certificate cert, AuthenticationContext authenticationContext)
     throws AuthenticationFailedException {
```

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java [hunk 2]
```diff
@@ -424,7 +424,8 @@ protected boolean retryAuthenticationEnabled() {
      * @param cert                  x509 certificate.
      * @param authenticationContext authenticationContext
      */
-    private String getMatchedAlternativeName(X509Certificate cert, AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+    protected String getMatchedAlternativeName(X509Certificate cert, AuthenticationContext authenticationContext)
+     throws AuthenticationFailedException {
```
Comment (piraveena) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java]:
```suggestion
             throws AuthenticationFailedException  {
```

**Comment by piraveena** [comment]
Sends this fix to 2.0.x branch with this PR https://github.com/wso2-extensions/identity-outbound-auth-x509/pull/42. Hence closing this PR

**Pull request closed without merging**